subprojects/docs/src/docs/userguide/publishing_maven.adoc

```diff
@@ -106,6 +106,30 @@ include::sample[dir="signing/maven-publish/groovy",files="build.gradle[tags=pom-
 include::sample[dir="signing/maven-publish/kotlin",files="build.gradle.kts[tags=pom-customization]"]
 ====
 
+[[publishing_maven:resolved_dependencies]]
+=== Customizing dependencies versions
+
+By default, the set of dependencies and constraints that are added to the POM file will contain the versions _declared_ in the build file.
+Sometimes it is desirable to publish the _resolved_ version instead.
+
+Here are a couple use cases for this:
+
+* A project leverages the rich versions constraints of Gradle, which have a lossy conversion to Maven.
+Instead of relying on the conversion, it publishes the resolved versions.
+* In combination with <<dependency_locking.adoc,dependency locking>>, you do not want to publish the dynamic versions but instead the locked ones.
```

Comment:
:x: I would put dependency locking first as it's a most frequent use case. I would also mention that even without locking it's interesting. For example, your build file contains `latest.release` or `1.+`, and you want the resolved version when publishing. You don't need locking for this.

Reply:
Funny, I had that order first and decided to flip it. Edited so that the dynamic version use case is first, followed by locking and then rich version constraints. That last point will get a link once #8047 is merged.